src/Compilers/CSharp/Test/Symbol/Symbols/SymbolErrorTests.cs

```diff
@@ -13818,8 +13819,10 @@ static int Main()
 }
 ";
             CreateStandardCompilation(text).VerifyDiagnostics(
-                // (9,20): error CS0842: 'TestNamespace.Str.Num': Automatically implemented properties cannot be used inside a type marked with StructLayout(LayoutKind.Explicit)
-                Diagnostic(ErrorCode.ERR_ExplicitLayoutAndAutoImplementedProperty, "Num").WithArguments("TestNamespace.Str.Num"));
+                // (9,20): error CS0625: 'Str.<Num>k__BackingField': instance field types marked with StructLayout(LayoutKind.Explicit) must have a FieldOffset attribute
+                //         public int Num // CS0625
+                Diagnostic(ErrorCode.ERR_MissingStructOffset, "Num").WithArguments("TestNamespace.Str.<Num>k__BackingField").WithLocation(9, 20)
```

Comment:
>Diagnostic(ErrorCode.ERR_MissingStructOffset, "Num").WithArguments("TestNamespace.Str.<Num>k__BackingField").WithLocation(9, 20) [](start = 16, length = 128)

It feels like we should be reporting a friendlier error, suggesting to add the attribute on the property with a field target. The name of the backing field probably shouldn't be mentioned at all.